### PR TITLE
[TRAH] shontzu/TRAH-3015/ui-for-seamless-MT5-mobile-login-integration

### PR DESCRIPTION
**Pull request by shontzu-deriv**
## Changes:

- this screen but for v2
- [CU](https://app.clickup.com/t/20696747/TRAH-3015)
- [figma](https://www.figma.com/file/9MJZgnyxjTQaXMcGyYvMO7/Release-(Rebuild)---High-risk-ROW-Trader%E2%80%99s-hub?node-id=14%3A28698&mode=dev)

image

https://github.com/deriv-com/traders-hub/assets/108507236/21b74cd9-99be-48bc-8e56-d7e74f1cb2cc

### CODE REVIEW

File: src/cfd/screens/TradeScreen/MT5MobileRedirectOption.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+            <Text as='p' size='2xs'>
+                Note: Don&apos;t have the MT5 app? Tap the <span>Trade with MT5 mobile app</span> button to download.
+                Once you have installed the app, return to this screen and hit the same button to log in.
+            </Text>
+        </div>
+    );
+};
+
+export default MT5MobileRedirectOption;
```
Comment (thisyahlen-deriv) [inline src/cfd/screens/TradeScreen/MT5MobileRedirectOption.tsx:61]: Use export const at the top pls. Lets not use this method anymore

File: src/cfd/constants/constants.tsx [hunk 2]
```diff
@@ -250,3 +251,20 @@ export const AppToIconMapper: Record<string, ComponentType<SVGAttributes<SVGElem
...
+
+export const getDeeplinkUrl = ({ details }: { details: THooks.MT5AccountsList }) => {
+    return `metatrader5://account?login=${details?.display_login}&server=${details?.server_info?.environment}`;
+};
+
+export const getMobileAppInstallerUrl = ({ details }: { details: THooks.MT5AccountsList }) => {
+    if (mobileOsDetect() === 'iOS') {
+        return details?.white_label_links?.ios;
+    } else if (mobileOsDetect() === 'huawei') {
+        return 'https://appgallery.huawei.com/#/app/C102015329';
```
Comment (hasan-deriv) [inline src/cfd/constants/constants.tsx:267]: Why this link is hard coded?

File: src/cfd/screens/TradeScreen/MT5MobileRedirectOption.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                <div className='flex justify-between w-full gap-5 align-center'>
+                    <IconComponent height={16} icon='Laptop' width={16} />
+                    <Text align='left' className='flex-1' size='xs' weight='bold'>
+                        MetaTrader5 web terminal
+                    </Text>
+                    <IconComponent height={16} icon='ChevronRight' width={16} />
+                </div>
+            </a>
+            <button
+                className='flex justify-between w-full gap-2 p-8 bg-blue-600 rounded-md fill-white align-center text-decoration-none'
```
Comment (hasan-deriv) [inline src/cfd/screens/TradeScreen/MT5MobileRedirectOption.tsx:41]: `fill-white` is not needed here.

File: src/cfd/screens/TradeScreen/TradeScreen.tsx [hunk 4]
```diff
@@ -138,25 +140,30 @@ export const TradeScreen = ({ account }: TradeScreenProps) => {
...
+                        </Fragment>
+                    ) : (
+                        <MT5MobileRedirectOption details={details as THooks.MT5AccountsList} />
+                    ))}
+
                 {platform === dxtradePlatform && (
-                    <TradeLink app={DesktopLinks.DXTRADE_WEB} platform={dxtradePlatform} />
+                    <TradeLink
+                        app={DesktopLinks.DXTRADE_WEB}
+                        // isDemo={activeAccount?.is_virtual}
```
Comment (hasan-deriv) [inline src/cfd/screens/TradeScreen/TradeScreen.tsx:164]: remove the comment 😉

File: src/cfd/constants/constants.tsx (hunk 2 above)
Comment (shontzu-deriv) [reply src/cfd/constants/constants.tsx:267]: unfortunately we are not getting all the required links needed from `white_label_links` 
will raise this

File: src/cfd/constants/constants.tsx [hunk 5]
```diff
@@ -250,3 +251,20 @@ export const AppToIconMapper: Record<string, ComponentType<SVGAttributes<SVGElem
...
+export const getWebtraderUrl = ({ details }: { details: THooks.MT5AccountsList }) => {
+    return `${details?.white_label_links?.webtrader_url}?login=${details?.display_login}&server=${details?.server_info?.environment}`;
+};
+
+export const getDeeplinkUrl = ({ details }: { details: THooks.MT5AccountsList }) => {
+    return `metatrader5://account?login=${details?.display_login}&server=${details?.server_info?.environment}`;
+};
+
+export const getMobileAppInstallerUrl = ({ details }: { details: THooks.MT5AccountsList }) => {
+    if (mobileOsDetect() === 'iOS') {
```
Comment (amina-deriv) [inline src/cfd/constants/constants.tsx:264]: do we have any shared constant for these different OS ?

Comment (shontzu-deriv) [reply src/cfd/constants/constants.tsx:264]: nope @amina-deriv , this snippet is taken from legacy code and in legacy code there is no constant for the OS

Comment (shontzu-deriv) [reply src/cfd/constants/constants.tsx:264]: we can use constants for the OS here

**Review by hasan-deriv — APPROVED** [review]

**Comment by shontzu-deriv** [comment]
please review the new `urlConfig` file I created

**Review by thisyahlen-deriv — APPROVED** [review]

**Comment by shontzu-deriv** [comment]
please review the new `urlConfig` file I created, this will be sent as a proposal to the BE

**Review by amina-deriv — APPROVED** [review]

**Pull request merged**